src/access/AccessControl.cpp

```diff
@@ -59,6 +61,26 @@ bool CheckRequestPrivilegeAgainstEntryPrivilege(Privilege requestPrivilege, Priv
     return false;
 }
 
+bool CheckSubjectDescriptorCATsAgainstEntrySubject(CATValues cats, NodeId subject)
+{
+    constexpr uint64_t kTagIdentifierMask = 0x0000'0000'FFFF'0000ULL;
+    constexpr uint64_t kTagVersionMask    = 0x0000'0000'0000'FFFFULL;
+
+    for (size_t i = 0; i < cats.size(); ++i)
```

Comment:
I believe you can use these kinds of loops:

```
for (auto cat : cats.val)
{
    // check cat...
}
```

Reply:
updated